external-crates/move/crates/move-compiler/src/linters/abort_constant.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Lint to encourage the use of named constants with 'abort' and 'assert' for enhanced code readability.
+//! Detects cases where non-constants are used and issues a warning.
+use crate::{
+    diag,
+    diagnostics::{
+        codes::{custom, DiagnosticInfo, Severity},
+        WarningFilters,
+    },
+    linters::{LinterDiagnosticCategory, ABORT_CONSTANT_DIAG_CODE, LINT_WARNING_PREFIX},
+    shared::CompilationEnv,
+    typing::{
+        ast::{self as T, BuiltinFunction_, ExpListItem, UnannotatedExp_},
+        visitor::{TypingVisitorConstructor, TypingVisitorContext},
+    },
+};
+use move_proc_macros::growing_stack;
+
+const ABORT_CONSTANT_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagnosticCategory::Style as u8,
+    ABORT_CONSTANT_DIAG_CODE,
+    "Prefer using named constants with 'abort' and 'assert' for clarity",
```

Comment:
The top-line messages are usually very succinct, and without capitalization where they are in the error 
```suggestion
    "use named constants with 'abort' and 'assert'",
```

Reply:
refactored